chainstate/test-framework/src/random_tx_maker.rs

```diff
@@ -700,10 +781,56 @@ impl<'a> RandomTxMaker<'a> {
                             }
                         }
 
-                        TxOutput::Transfer(
-                            OutputValue::Coin(new_value),
-                            Destination::AnyoneCanSpend,
-                        )
+                        let destination =
+                            key_manager.new_destination(self.chainstate.get_chain_config(), rng);
+                        match rng.gen_range(0..5) {
+                            0 => TxOutput::LockThenTransfer(
+                                OutputValue::Coin(new_value),
+                                destination,
+                                OutputTimeLock::ForBlockCount(rng.gen_range(0..5)),
+                            ),
+                            1 => TxOutput::LockThenTransfer(
+                                OutputValue::Coin(new_value),
+                                destination,
+                                OutputTimeLock::UntilHeight(
+                                    self.chainstate
+                                        .get_best_block_height()
+                                        .unwrap()
+                                        .checked_add(rng.gen_range(0..5))
+                                        .unwrap(),
+                                ),
+                            ),
+                            2 => TxOutput::LockThenTransfer(
+                                OutputValue::Coin(new_value),
+                                destination,
+                                OutputTimeLock::ForSeconds(
+                                    self.chainstate
+                                        .get_chain_config()
+                                        .target_block_spacing()
+                                        .as_secs()
+                                        * rng.gen_range(0..5),
```

Comment:
Seems it would be better to scale the interval by the block spacing and generate number of seconds with one second resolution rather than multiples of `target_block_spacing`. Same for `LockThenTransfer` below.

Reply:
Agreed. This limitation is unnecessary. I can also live with keeping this like this if we add a random offset in the range [0, target_spacing]. This would ensure that we cover all possible values.